src/java.base/share/classes/java/util/HexFormat.java

```diff
@@ -173,29 +165,37 @@ public final class HexFormat {
      * The hexadecimal characters are from lowercase alpha digits.
      */
     private static final HexFormat HEX_FORMAT =
-            new HexFormat("", "", "", LOWERCASE_DIGITS);
+            new HexFormat("", "", "", Case.LOWERCASE);
+
+    private static final HexFormat HEX_UPPER_FORMAT =
+            new HexFormat("", "", "", Case.UPPERCASE);
 
     private static final byte[] EMPTY_BYTES = {};
 
     private final String delimiter;
     private final String prefix;
     private final String suffix;
-    private final byte[] digits;
+    private final Case digitCase;
+
+    private enum Case {
+        LOWERCASE,
+        UPPERCASE
+    }
```

Comment:
Instead of an enum cache the "A" or "a" for direct use.
`private final char caseBase;`
Initialize it in the constructor.